proto/pbconfigentry/config_entry.proto

```diff
@@ -544,9 +546,9 @@ message UpstreamLimits {
 message PassiveHealthCheck {
   // mog: func-to=structs.DurationFromProto func-from=structs.DurationToProto
   google.protobuf.Duration Interval = 1;
-  uint32 MaxFailures = 2;
+  optional uint32 MaxFailures = 2;
```

Comment:
Why did these become `optional`? https://github.com/hashicorp/consul/pull/15614/files#diff-9d2e9c8b34664ef85c83b6c6f4ec7f939cce7a64f41c267e5db248c1f95b0571R549-R551

Do you know if this is a breaking change?

Reply:
IMO, this is not a breaking change.

Optional makes `mog` know if `MaxFailures` is defined or not. Otherwise, unspecified `MaxFailures` will become "0", which is a wrong value. In the PR, I added a test for unspecified `EnforcingConsecutive5xx`, which is the same type as `MaxFailures`, to verify the default value is "null" for envoy.

https://github.com/hashicorp/consul/pull/15614/files#diff-a44e2ceb5ced90fcc9be4d9518bb62626d62438751d9cc67286de9df15d15b54R72

Reply:
@erichaberkorn , there is a test that asserts the `maxFailure` in upstream config

https://github.com/hashicorp/consul/blob/4a32070210cdbff8c457334ee2e3e798b9f263b1/test/integration/connect/envoy/case-upstream-config/verify.bats#L49-L52

Reply:
Another test is added to verify unspecified `MaxFailure` in `upstreams.config.passive_health_check`

https://github.com/hashicorp/consul/pull/15614/commits/30af125d96bb7b6665659c428b7f62b59df193df

Reply:
I'm talking about a protobuf compatibility breaking change. I'm pretty sure we would need to introduce a completely different field on the protobuf and deprecate the current field to get this working. Then we could remove the current field in a follow-up release.

Reply:
@erichaberkorn , reverted the change to `message PassiveHealthCheck` by adding the constraint of `EnforcingConsecutive5xx > 0`. WDYT?